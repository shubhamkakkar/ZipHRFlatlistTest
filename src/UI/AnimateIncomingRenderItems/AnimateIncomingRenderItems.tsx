import React, { useEffect } from 'react';
import { Animated, Dimensions } from 'react-native';
import styles from './styleSheetAnimateIncomingRenderItems';

type TAnimateIncomingRenderItems = {
    index: number;
    trigger: boolean;
    children: React.ReactNode;
}
const { height: HEIGHT } = Dimensions.get('screen');

export default function AnimateIncomingRenderItems(
  { index, trigger, children }: TAnimateIncomingRenderItems,
) {
  const translateY = new Animated.Value(HEIGHT);
  useEffect(() => {
    if (trigger) {
      Animated.timing(translateY, {
        toValue: 0,
        delay: index < 5 ? index * 10 : 10,
        useNativeDriver: true,
        duration: 250,
      }).start();
    }
  }, [trigger]);

  return (
    <Animated.View
      style={[
        styles.container,
        {
          transform: [
            {
              translateY,
            },
          ],
        },
      ]}
    >
      {children}
    </Animated.View>
  );
}
