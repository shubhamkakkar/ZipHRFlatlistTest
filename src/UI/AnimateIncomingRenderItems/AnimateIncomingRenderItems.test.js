import { View } from 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import AnimateIncomingRenderItems from './AnimateIncomingRenderItems';

beforeEach(() => {
  jest.useFakeTimers();
});

it('AnimateIncomingRenderItems matches snapshot - props: {{ index: 0, trigger: false, }}', async () => {
  jest.advanceTimersByTime(500);
  const tree = await renderer.create(
    <AnimateIncomingRenderItems {...{
      index: 0,
      trigger: false,
    }}
    >
      <View />
    </AnimateIncomingRenderItems>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('AnimateIncomingRenderItems matches snapshot - props: {{ index: 0, trigger: true, }}', async () => {
  jest.advanceTimersByTime(500);
  const tree = await renderer.create(
    <AnimateIncomingRenderItems {...{
      index: 0,
      trigger: true,
    }}
    >
      <View />
    </AnimateIncomingRenderItems>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
