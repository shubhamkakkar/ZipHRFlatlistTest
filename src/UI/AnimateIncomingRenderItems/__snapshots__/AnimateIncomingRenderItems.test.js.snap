// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimateIncomingRenderItems matches snapshot - props: {{ index: 0, trigger: false, }} 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "transform": Array [
        Object {
          "translateY": 1334,
        },
      ],
    }
  }
>
  <View />
</View>
`;

exports[`AnimateIncomingRenderItems matches snapshot - props: {{ index: 0, trigger: true, }} 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "transform": Array [
        Object {
          "translateY": 1334,
        },
      ],
    }
  }
>
  <View />
</View>
`;
